### ISSUE TITLE
Fix emf2svg build error on Ruby 3.1 with Windows (bundle install)

### ISSUE THREAD

**Issue by ronaldtse** [metanorma/isodoc#385]
There is a build error on Ruby 3.1 with Windows about bundle install:

```
CMake Error at vcpkg/scripts/buildsystems/vcpkg.cmake:827 (message):
    vcpkg install failed.  See logs for more information:
    D:\a\_temp\d20220314-4988-dnd7iw\libemf2svg\vcpkg-manifest-install.log
  Call Stack (most recent call first):
  C:/Program Files/CMake/share/cmake-3.22/Modules/CMakeDetermineSystem.cmake:121
  (include)
    CMakeLists.txt:17 (project)
  
  
  CMake Error at CMakeLists.txt:17 (project):
    Generator
  
      Visual Studio 16 2019
  
    could not find any instance of Visual Studio.
```

_Originally posted by @ronaldtse in https://github.com/metanorma/isodoc/issues/384#issuecomment-1067509501_

**Comment by ronaldtse** [comment]
@maxirmx any idea on why this build fails?

https://github.com/metanorma/isodoc/runs/5545183883?check_suite_focus=true

**Comment by ronaldtse** [comment]
@opoudjis is also seeing such failure across the board: https://github.com/metanorma/metanorma-release

**Comment by CAMOBAP** [comment]
`emf2svg-ruby` is just a wrapper gem for the native `emf2svg` library which loaded via FFI.

So it doesn't require a separate version of a library for each Ruby MAJOR.MINOR version.

So the question why did bundler decide to not use the prebuilt x64-mingw32 but get gem with sources to compile.

I previously faced such an issue on macOS and it was related to the calculation of platform label (i.e. `64-mingw32`). Maybe for Windows 3.1 we have something similar. Investigating ...

**Comment by maxirmx** [comment]
As far as I remember emf2svg library employs native (Microsoft) toolset.  It fails saying that it cannot find Microsoft compiler.

**Comment by CAMOBAP** [comment]
> As far as I remember emf2svg library employs native (Microsoft) toolset. It fails saying that it cannot find Microsoft compiler.

This is true, and to avoid building complexity, we can provide precompiled platform gem. As you may see here https://rubygems.org/gems/emf2svg/versions we have 4 gems for the same version 1.0.0:
 - [1.0.0](https://rubygems.org/gems/emf2svg/versions/1.0.0-x64-mingw32) - September 06, 2021 x64-mingw32 (1.61MB)
 - [1.0.0](https://rubygems.org/gems/emf2svg/versions/1.0.0-x86_64-darwin) - September 06, 2021 x86_64-darwin (163KB)
 - [1.0.0](https://rubygems.org/gems/emf2svg/versions/1.0.0-x86_64-linux) - September 06, 2021 x86_64-linux (217KB)
 - [1.0.0](https://rubygems.org/gems/emf2svg/versions/1.0.0) - September 06, 2021 (12KB)

First 3 are precompiled so they come with compiled shared libraries.

Basically on the bundler's side during resolving dependencies it should look for precompiled gem first  by comparing `platform` (`x64-mingw32`, `x86_64-darwin`, `x86_64-linux`)

So in theory this can happens for windows on arm or x86, but I'm not sure that this is our case

I will continue investigation

**Comment by maxirmx** [comment]
**Addition re Microsoft compiler**
emf2svg build script explicitly references `MS Visual Studio 2019`  That was/is relevant for `Windows Server 2019` GHA runner
Now `windows-latest` is `Windows Server 2022` that has `MS Visual Studio 2022`
That's why build step fails

**Comment by maxirmx** [comment]
It does not use `x64-mingw32` precompiled version because the GHA runner is configured for `x64-ucrt64`
Simply speaking gen is pre-built with gcc and GHA deploys llvm

**Comment by CAMOBAP** [comment]
@maxirmx BTW is it possible to specify 'minimal' version of Genetator for cmake?

**Comment by ronaldtse** [comment]
Apparently this is a known issue in Rubygems: https://github.com/rubygems/rubygems/issues/5269

> The platforms x64-mingw32 and x64-mingw-ucrt are ABI incompatible - that's why we introduced a new platform string for RUBY_PLATFORM and Rubygems

Can we build a precompiled version for `x64-mingw-ucrt` for all our Windows gems?

**Comment by maxirmx** [comment]
> @maxirmx BTW is it possible to specify 'minimal' version of Genetator for cmake?

The library itself uses the configuration provided by the runner
```
       cmake 
        -B ${{github.workspace}} 
        -DCMAKE_BUILD_TYPE=${{env.BUILD_TYPE}} 
        -DCMAKE_TOOLCHAIN_FILE=${{github.workspace}}/vcpkg/scripts/buildsystems/vcpkg.cmake
```
so it shall be version-agnostic.   
I have to look at gem build script to see why and where  VS version is hardcoded

**Comment by ronaldtse** [comment]
@maxirmx it is hardcoded here: https://github.com/metanorma/emf2svg-ruby/blob/4d7d99717f1daefba73a18da3b5b5c693f7edc82/lib/emf2svg/recipe.rb#L42

**Comment by ronaldtse** [comment]
Thank you guys -- let's move the discussion to https://github.com/metanorma/emf2svg-ruby/issues/7 instead since it is more relevant there.

**Comment by ronaldtse** [comment]
Thanks!
